library/src-bootstrapped/scala/internal/quoted/Type.scala

```diff
@@ -47,30 +47,48 @@ object Type {
   }
 
   def Unit: QuoteContext ?=> quoted.Type[Unit] =
-    qctx.tasty.defn.UnitType.seal.asInstanceOf[quoted.Type[Unit]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_UnitType.seal.asInstanceOf[quoted.Type[Unit]]
+
 
   def Boolean: QuoteContext ?=> quoted.Type[Boolean] =
-    qctx.tasty.defn.BooleanType.seal.asInstanceOf[quoted.Type[Boolean]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_BooleanType.seal.asInstanceOf[quoted.Type[Boolean]]
+
 
   def Byte: QuoteContext ?=> quoted.Type[Byte] =
-    qctx.tasty.defn.ByteType.seal.asInstanceOf[quoted.Type[Byte]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_ByteType.seal.asInstanceOf[quoted.Type[Byte]]
+
 
   def Char: QuoteContext ?=> quoted.Type[Char] =
-    qctx.tasty.defn.CharType.seal.asInstanceOf[quoted.Type[Char]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_CharType.seal.asInstanceOf[quoted.Type[Char]]
+
 
   def Short: QuoteContext ?=> quoted.Type[Short] =
-    qctx.tasty.defn.ShortType.seal.asInstanceOf[quoted.Type[Short]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_ShortType.seal.asInstanceOf[quoted.Type[Short]]
+
 
   def Int: QuoteContext ?=> quoted.Type[Int] =
-    qctx.tasty.defn.IntType.seal.asInstanceOf[quoted.Type[Int]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_IntType.seal.asInstanceOf[quoted.Type[Int]]
+
 
   def Long: QuoteContext ?=> quoted.Type[Long] =
-    qctx.tasty.defn.LongType.seal.asInstanceOf[quoted.Type[Long]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_LongType.seal.asInstanceOf[quoted.Type[Long]]
+
 
   def Float: QuoteContext ?=> quoted.Type[Float] =
-    qctx.tasty.defn.FloatType.seal.asInstanceOf[quoted.Type[Float]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_FloatType.seal.asInstanceOf[quoted.Type[Float]]
+
 
   def Double: QuoteContext ?=> quoted.Type[Double] =
-    qctx.tasty.defn.DoubleType.seal.asInstanceOf[quoted.Type[Double]]
+    val qctx1 = quoteContextWithCompilerInterface(qctx)
+    qctx1.tasty.Definitions_DoubleType.seal.asInstanceOf[quoted.Type[Double]]
+
```

Comment:
Can we do the same for `Type.Int`, etc as in `Reflection.Type`?

Reply:
That definition is not visible in the public API. It is used as an optimization of `'[Int]`. I'm looking for other ways to optimize it differently but it will take some time.

Reply:
I see, it makes sense.